src/field-value.js

```diff
@@ -57,7 +45,7 @@ class FieldValue {
    * });
    */
   static delete() {
-    return DELETE_SENTINEL;
+    return new DeleteTransform();
```

Comment:
Kinda minor, so feel free to ignore, but you could have a singleton for the transform instance, right? Instead of allocating a new one each time it's used?

Reply:
I reverted this back to the way it was before and added singletons.